karyon3-api/src/main/java/com/netflix/karyon/api/health/HealthIndicatorStatus.java

```diff
@@ -7,14 +7,45 @@
 import java.util.Map.Entry;
 
 /**
- * Utility class for creating common {@link HealthIndicatorStatus} instances
+ * Immutable health check instance returned from a {@link HealthIndicator}
  * 
+ * See {@link HealthIndicatorStatus} for utility methods to create HealthIndicatorStatus objects
  * @author elandau
- *
  */
-public final class HealthIndicatorStatuses {
+public interface HealthIndicatorStatus {
```

Comment:
I know this is more a meta question than a critique of this move(which I generally find to be a good idea), but should we really even be doing this much stuff?  Why isn't a health check just a true/false, and why are we so involved in this process and defining complex APIs?